pkg/gadgets/run/types/types.go

```diff
@@ -85,10 +94,56 @@ type Printer interface {
 	Logf(severity logger.Level, fmt string, params ...any)
 }
 
+// Type represents the type of a field.
+
+type Kind uint8
+
+// TODO: use reflect.Kind? (once we resolve TODO below)
+const (
+	KindNone = iota
+	KindUint8
+	KindUint16
+	KindUint32
+	KindUint64
+	KindInt8
+	KindInt16
+	KindInt32
+	KindInt64
+	KindBool
+	KindFloat32
+	KindFloat64
+	KindString
+	KindArray
+
+	// TODO: this should be made more generic to avoid coupling operators with this
+	KindL3Endpoint
+	KindL4Endpoint
+	KindTimestamp
+)
+
+type Type struct {
+	Kind Kind
+
+	// Fields only for KindArray
+	ArrayNElements int // number of elements in the array
+	ArrayType      *Type
+}
+
+// ColumnDesc describes how a column is built. It's basically a serializable version of
+// columns.DynamicField
+type ColumnDesc struct {
+	Name      string
+	BlobIndex int // -1: virtual, 0: ebpf, 1: fixed length, 1+ strings
```

Comment:
Would it make it clearer to split this into two fields?

```suggestion
	BlobType BlobType // BlobType is an enum defined with iotas
	StringIndex int // only used when BlobType == BlobTypeString
```